default.nix

```diff
@@ -96,24 +98,68 @@ let
       ghc = pkgs.haskell.compiler.ghc843;
     };
 
+    playgroundGhc = pkgs.haskell.packages.ghc843.ghcWithPackages (ps: [
```

Comment:
This seems a bit weird. It should certainly use the compiler from the package set we've already constructed, rather than specifying 843 here.

Reply:
it is the same compiler `ghc = pkgs.haskell.compiler.ghc843;` on line 98. `ghcWithPackages` just wraps up that compiler with the libraries we want hlint to have access to.

Reply:
We shouldn't be specifying it there either, IMO that's a bug in iohk-nix. Let's not add another place. Just get it out of the package set.

Ah, I finally get it: you need a GHC at *runtime* for `hint`. Regardless, more explanation of all these pieces in the code would be good.

Reply:
Oh, and indeed we could do that on line 98 too, I should do that.

Reply:
No, that was for a previous version of `iohk-nix` where it was accidentally hardcoding the compiler in a couple of places unless you passed it in.

From the point of view of everything in this file, no choice other than "use the same compiler as the main package set" is ever going to work, so let's just do that. At the moment this is just another place to make a mistake.

Reply:
fixed